### DISCUSSION TITLE
Autocomplete opens in (and moves input focus to) the wrong window

### DISCUSSION THREAD

**Discussion by ncallaway** [neoclide/coc.nvim#4578] [Q&A] [answered]
When using vim with coc installed, auto-complete is opening and causing characters to be inserted into the incorrect window.

To reproduce reliably I setup:
- A directory containing two files (`main.ts` and `second.ts`).
- Open the first file in vim: `vi main.ts`
- Do any editing which triggers auto-complete (e.g. enter insert mode type 'i', which prompts auto-complete option of `if`, among others)
- Open the `second.ts` file in another window: `:vsplit second.ts`
- In the new window (from now on the _right_ window, and the window showing `main.ts` will be the _left_ window), I do any editing which triggers auto-complete (e.g. enter insert mode and type 'h')
  - The autocomplete options appear in the _left_ window
  - Any subsequent characters are also entered in the _left_ window (e.g. if I type "hello", `h` is entered in the _right_ window, and `ello` is entered in the _left_ window.
- Any characters incorrectly entered in the _left_ window, are part of the buffer (e.g. if I save the left window with `:w`, `ello` will be included in the file written to disk)
- Those same edits are _not_ part of the undo history for that buffer (e.g. if after typing "hello", I exit insert mode and `U` to undo, the characters `ello` remain, and the previous edit in the _left_ window is undone).

Below is a short screen recording demonstrating the above:
https://user-images.githubusercontent.com/427497/226716774-605e0e12-5f1c-4126-9b68-3713c9d03e87.mp4

I'm using vim `VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Nov 21 2022 00:00:00)`, and installed coc through plug (`Plug 'neoclide/coc.nvim', {'branch': 'release'}`). The issue goes away if I comment out `coc` from my plug section. I'm happy to share my `.vimrc` or any other configuration files that are helpful in diagnosing the issue.

**Comment by fannheyward** [comment, marked as answer]
> from now on the right window, and the window showing main.ts will be the left window

In my tests, the `main.ts` is in the right window, tested with both nvim and vim 9.0 9001400.

**Reply by ncallaway** [reply]
@fannheyward Ah, yea, sorry, I forgot that in my `.vimrc` file I have `set splitright`, which inverts the default positioning of the vsplit.

I've confirmed that the issue also occurs without `:set splitright` (where `main.ts` is on the right, and `second.ts` is on the left). But the same issue occurs.

Are there any useful diagnostics or configuration information I can provide?

**Reply by fannheyward** [reply]
Can't reproduce your issue, completions opens in the correct window. Try minimal vimrc, with coc.nvim only.

**Reply by ncallaway** [reply]
@fannheyward 

Here's the minimal vimrc I created for this.

```
$ cat ~/.vimrc
call plug#begin()
Plug 'neoclide/coc.nvim', {'branch': 'release'}
call plug#end()
```

The issue is still occurring for me with that vimrc. If I have _no_ vimrc, the issue doesn't occur. Let_ me know if there's anything else I can try and eliminate or remove to narrow down.

Here's a link to a screen recording with _no_ vimrc, where the issue does not occur:  https://user-images.githubusercontent.com/427497/227057972-f5422b58-f600-44d0-a622-3c6a421cceaa.mp4

And here's a link to a screen recording with the minimal vimrc listed above, where the issue does occur: https://user-images.githubusercontent.com/427497/227058093-048fb5b8-77c3-496d-9a45-919725fdd650.mp4

I'm very new to both vim and coc, so if there's any other environmental information I can provide, or anything else I can eliminate that would be helpful let me know.

**Reply by fannheyward** [reply]
Similar issue https://github.com/neoclide/coc.nvim/issues/4396, upgrade your Vim and try again.

**Reply by ncallaway** [reply]
@fannheyward That fixed it! Thank you!

**Reply by ncallaway** [reply]
I apparently can't mark this comment as the "answer", so I've marked the top level thread as the answer.
